rllib/evaluation/collectors/sample_collector.py

```diff
@@ -1,9 +1,10 @@
 from abc import abstractmethod, ABCMeta
 import logging
-from typing import Dict, Optional
+from typing import Dict, Union
```

Comment:
Re-implemented SampleCollector (as a basic list-based solution, which is easier to maintain and not slower than the purely pre-allocated-buffer strategy we had previously).